### PR TITLE
Add `external-snapshot-metadata` repo to CSI jobs

### PR DESCRIPTION
**Pull request by hairyhum**
Added `external-snapshot-metadata` to `config/jobs/kubernetes-csi/gen-jobs.sh` and `config/jobs/image-pushing/k8s-staging-sig-storage.sh` and generated the jobs.

Added `external-snapshot-metadata` to `config/testgrids/kubernetes/sig-storage/config.yaml` dashboards.

This is a part of setting up the CI for https://github.com/kubernetes-csi/external-snapshot-metadata

Related:
https://github.com/kubernetes-csi/external-snapshot-metadata/issues/19
https://github.com/kubernetes-csi/external-snapshot-metadata/pull/6

I might have missed or misconfigured something, so would appreciate any feedback.

### CODE REVIEW

**Comment by hairyhum** [comment]
@xing-yang can you please take a look?

**Comment by hairyhum** [comment]
@xing-yang all tests look green, can you do a review?

File: config/jobs/kubernetes-csi/external-snapshot-metadata/external-snapshot-metadata-config.yaml [hunk 1]
```diff
@@ -0,0 +1,433 @@
+# generated by gen-jobs.sh, do not edit manually
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/external-snapshot-metadata/external-snapshot-metadata-config.yaml:1]: Is this file generated by running gen-jobs.sh?  We recently updated K8s versions in gen-jobs.sh.  Please re-run it.  We should have jobs for 1.29-1.31.

Comment (hairyhum) [reply config/jobs/kubernetes-csi/external-snapshot-metadata/external-snapshot-metadata-config.yaml:1]: Regenerated.

**Comment by hairyhum** [comment]
@xing-yang ready for another review

**Comment by hairyhum** [comment]
@xing-yang @saad-ali @mpherman2 can this move forward? Anything else required from my side?

**Comment by xing-yang** [comment]
@dims Can you help approve this?
/assign @dims

File: config/jobs/image-pushing/k8s-staging-sig-storage.yaml [hunk 2]
```diff
@@ -321,6 +321,46 @@ postsubmits:
...
+      annotations:
+        testgrid-dashboards: sig-storage-image-build, sig-k8s-infra-gcb
+      decorate: true
+      decoration_config:
+        timeout: 240m
+        grace_period: 15m
+      branches:
+        # For publishing canary images.
+        - ^master$
+        - ^release-
```
Comment (Rakshith-R) [inline config/jobs/image-pushing/k8s-staging-sig-storage.yaml:344]: Hey @hairyhum @xing-yang ,

The development branch for https://github.com/kubernetes-csi/external-snapshot-metadata is `main` and not master `master` as with other sidecars. 

The build job will not get triggered due to this difference ? 

https://github.com/kubernetes-csi/external-snapshot-metadata 
I don't this job being triggered for the pr that merged 3 days ago.

cc @PrasadG193 @carlbraganza

**Pull request merged**